packages/shared/lib/services/endUser.service.ts

```diff
@@ -171,3 +171,76 @@ export async function getEndUserByConnectionId(db: Knex, props: { connectionId:
 
     return Ok(EndUserMapper.from(endUser));
 }
+
+export async function upsertEndUser(
+    db: Knex,
+    {
+        account,
+        environment,
+        endUserPayload,
+        organization
+    }: { account: DBTeam; environment: DBEnvironment; endUserPayload: ConnectSessionInput['end_user']; organization?: ConnectSessionInput['organization'] }
+): Promise<Result<EndUser, EndUserError>> {
+    // Check if the endUser exists in the database
```

Comment:
can we not use pg `INSERT ON CONFLICT` to handle the upsert?

Reply:
I just copied your code actually, we can note to improve that for sure